sources/packages/api/src/services/education-program-offering/education-program-offering-import-csv.service.ts

```diff
@@ -0,0 +1,227 @@
+    private readonly educationProgramService: EducationProgramService,
+  ) {}
+
+  /**
+   * Generates the offering save model from the CSV content, including
+   * any additional information like the program information and location
+   * information needed to execute the complete offering validation.
+   * @param institutionId institution id that will receive the offerings.
+   * @param csvModels CSV models to be converted to the offering models.
+   * @returns offering models to be validated and persisted.
+   */
+  async generateSaveOfferingModelFromCSVModels(
+    institutionId: number,
+    csvModels: OfferingCSVModel[],
+  ): Promise<SaveOfferingModel[]> {
+    // Locations and programs information to be associated with the offering models.
+    const [locationsMap, programsMap] = await Promise.all([
+      this.getLocationsMaps(institutionId),
+      this.getProgramsMaps(institutionId, csvModels),
+    ]);
+    return csvModels.map((csvModel) => ({
+      offeringName: csvModel.offeringName,
+      yearOfStudy: csvModel.yearOfStudy,
+      showYearOfStudy: csvModel.showYearOfStudy === YesNoOptions.Yes,
+      offeringIntensity: csvModel.offeringIntensity,
+      offeringDelivered: csvModel.offeringDelivered,
+      hasOfferingWILComponent:
+        csvModel.WILComponent === YesNoOptions.Yes
+          ? WILComponentOptions.Yes
+          : WILComponentOptions.No,
+      offeringWILComponentType: csvModel.WILComponentType,
+      studyStartDate: csvModel.studyStartDate,
+      studyEndDate: csvModel.studyEndDate,
+      lacksStudyBreaks: csvModel.hasStudyBreaks == YesNoOptions.No,
+      studyBreaks: csvModel.studyBreaks,
+      actualTuitionCosts: csvModel.actualTuitionCosts,
+      programRelatedCosts: csvModel.programRelatedCosts,
+      mandatoryFees: csvModel.mandatoryFees,
+      exceptionalExpenses: csvModel.exceptionalExpenses,
+      offeringType:
+        csvModel.publicOffering === YesNoOptions.Yes
+          ? OfferingTypes.Public
+          : OfferingTypes.Private,
+      offeringDeclaration: csvModel.consent === YesNoOptions.Yes,
+      courseLoad: csvModel.courseLoad,
+      locationId: locationsMap[csvModel.institutionLocationCode],
+      programContext: programsMap[csvModel.sabcProgramCode],
+    }));
+  }
+
+  /**
+   * Get a list of all institution locations indexed by the institution code.
+   * @param institutionId institution to have all the locations retrieved.
+   * @returns list of all institution locations indexed by the institution code.
+   */
+  private async getLocationsMaps(
+    institutionId: number,
+  ): Promise<InstitutionCodeToIdMap> {
+    const locations = await this.institutionLocationService.getLocations(
+      institutionId,
+    );
+    const institutionMap: InstitutionCodeToIdMap = {};
+    locations.forEach((location) => {
+      if (location.institutionCode) {
+        institutionMap[location.institutionCode] = location.id;
```

Comment:
is the 4 digit institution code, if so, why its assigning as location id

Reply:
The assignment here will create an object that will map institutions codes and ids, like below:
```ts
{
    ABCD: 985,
    XPTO: 1234,
    KKBB: 6874
}
````